tool/tctl/common/plugin/okta.go

```diff
@@ -0,0 +1,268 @@
+		}
+		if !s.userSync {
+			return trace.BadParameter("User sync is required for AccessList sync")
+		}
+	}
+	if s.scimEnabled {
+		if s.scimToken == "" {
+			var err error
+			s.scimToken, err = utils.CryptoRandomHex(32)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+			s.autoGeneratedSCIMToken = true
+		}
+	}
+	connector, err := args.authClient.GetSAMLConnector(ctx, s.samlConnector, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if s.appID == "" {
+		appID, ok := connector.GetMetadata().Labels[types.OktaAppIDLabel]
+		if ok {
+			s.appID = appID
+		}
+	}
+	if s.scimToken != "" && s.appID == "" && s.userSync {
+		msg := []string{
+			"SCIM support requires App ID, which was not supplied and couldn't be deduced from the SAML connector",
+			"Specify the App ID explicitly with --app-id",
+			"SCIM support requires app-id to be set",
+		}
+		return trace.BadParameter(strings.Join(msg, "\n"))
+	}
+	return nil
+}
+
+func (p *PluginsCommand) InstallOkta(ctx context.Context, args installPluginArgs) error {
+	oktaSettings := p.install.okta
+	if err := oktaSettings.validateAndCheckDefaults(ctx, &args); err != nil {
+		return trace.Wrap(err)
+	}
+	creds, err := generateSCIMCreds(p.install.name, oktaSettings)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	settings := &types.PluginOktaSettings{
+		OrgUrl: oktaSettings.org.String(),
+		SyncSettings: &types.PluginOktaSyncSettings{
+			SsoConnectorId:       oktaSettings.samlConnector,
+			AppId:                oktaSettings.appID,
+			SyncUsers:            oktaSettings.userSync,
+			SyncAccessLists:      oktaSettings.accessListSync,
+			DefaultOwners:        oktaSettings.defaultOwners,
+			GroupFilters:         oktaSettings.groupFilters,
+			AppFilters:           oktaSettings.appFilters,
+			AppGroupSyncDisabled: !oktaSettings.appGroupSync,
+		},
+	}
+	req := &pluginsv1.CreatePluginRequest{
+		Plugin: &types.PluginV1{
+			SubKind: types.PluginSubkindAccess,
+			Metadata: types.Metadata{
+				Labels: map[string]string{
+					types.HostedPluginLabel: "true",
+				},
+				Name: p.install.name,
+			},
+			Spec: types.PluginSpecV1{Settings: &types.PluginSpecV1_Okta{Okta: settings}},
+		},
+		StaticCredentialsList: creds,
+		CredentialLabels: map[string]string{
+			types.OktaOrgURLLabel: oktaSettings.org.String(),
+		},
+	}
+
+	if _, err := args.plugins.CreatePlugin(ctx, req); err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Printf("Successfully created OKTA plugin %q\n\n", p.install.name)
+	if oktaSettings.scimEnabled {
+		pingResp, err := args.authClient.Ping(ctx)
+		if err != nil {
+			return trace.Wrap(err, "failed fetching cluster info")
+		}
+		scimBaseURL := fmt.Sprintf("https://%s/v1/webapi/scim/%s", pingResp.GetProxyPublicAddr(), p.install.name)
+		fmt.Printf("SCIM Base URL: %s\n", scimBaseURL)
+		fmt.Printf("SCIM Identifier field for users: %s\n", "userName")
+		if oktaSettings.autoGeneratedSCIMToken {
+			fmt.Printf("SCIM Baerar Token: %s\n", oktaSettings.scimToken)
```

Comment:
Was dubious about this feature until I saw that it only prints out tokens it generates. Nice.